### ISSUE TITLE
Two spell checker questions that no one ever thinks to ask or answer

### ISSUE THREAD

**Issue by TBugReporter** [Predelnik/DSpellCheck#82]
Any good spell checker will allow a user to _add_ words (uncommon technical terms, etc.), thereby designating words that are not found in its dictionary as nonetheless acceptable, but:
- How does a user undo this _add_ action if it gets done by mistake, or if it turns out to cause more problems than it solves?  And,
- How does a user _remove_ words from the built-in dictionary, for when it accepts rare words that, in the user's experience, are more likely to be misspellings of other words?

**Comment by janschreiber** [comment]
I agree with @TBugReporter that these two questions do not get enough attention traditionally.
But as long as computer-aided spell checking exists, the **first problem** could be solved by editing the user dictionary, which is virtually always a plain text file. The only difficulty here is to know where that file is located, and the location of the user dictionary is often not made fully transparent to the users.
LibreOffice already addresses the **second problem**: It allows you to blacklist certain words. See https://help.libreoffice.org/Common/Writing_Aids#Exceptions_.28-.29

**Comment by TBugReporter** [comment]
Thank you for your comment, but I was really hoping for some information specific to DSpellCheck, or rather to the DSpellCheck plugin for Notepad++.

**Comment by Hiro5** [comment]
It seems that the words that you add to dictionary, 
goes to the "<language>.usr" files, in this folder:
%APPDATA%\Notepad++\plugins\config\Hunspell\

**Comment by einst3** [comment]
remember that the *.usr files first line is a count of the words in the list. so if you remove one you must drement the count and if you add you must increment the count. 
best way is to review the file is in notepad (not npp). make your word changes, then in the view menu view the status bar, go down to the last word and put your cursor on that line, the statusbar will show Ln: #, Col 1. the #-1 is the number of words, goto the top and type that number on the first line. save file and your good to go.

**Comment by Predelnik** [comment]
From the looks of it Hunspell allows removal of word from its dictionary in run-time so blacklist may possibly be implemented. And removal of words from user dictionaries without editing text file could be done too. The only thing I'm cautious about is adding actions to context menu for correct words, maybe I'll implement them as additional hotkey.

**Comment by einst3** [comment]
how about adding an Edit User Dictionary context menu item to open separate window with list of items in user dictionary to add/edit/delete.

**Comment by Predelnik** [comment]
Yeah I think that's good idea too. I'll see what I can do.
